src/radical/entk/appman/wfprocessor.py

```diff
@@ -100,145 +110,173 @@ def _initialize_workflow(self):
-                                               new_state=states.SCHEDULING,
-                                               channel=mq_channel,
-                                               queue='%s-enq-to-sync' % self._sid,
-                                               profiler=local_prof,
-                                               logger=self._logger)
-
-                                executable_tasks = executable_stage.tasks
-
-                                for executable_task in executable_tasks:
-
-                                    if (executable_task.state == states.INITIAL)or \
-                                            ((executable_task.state == states.FAILED)and(self._resubmit_failed)):
-
-                                        # Set state of Tasks in current Stage to SCHEDULING
-                                        transition(obj=executable_task,
-                                                   obj_type='Task',
-                                                   new_state=states.SCHEDULING,
-                                                   channel=mq_channel,
-                                                   queue='%s-enq-to-sync' % self._sid,
-                                                   profiler=local_prof,
-                                                   logger=self._logger)
-
-                                        # task_as_dict = json.dumps(executable_task.to_dict())
-                                        workload.append(executable_task)
-
-                                        if executable_stage not in scheduled_stages:
-                                            scheduled_stages.append(
-                                                executable_stage)
-
+                        # If Pipeline is in the final state or suspended, we
+                        # skip processing it.
+                        if pipe.state in states.FINAL or  \
+                           pipe.completed or \
+                           pipe.state == states.SUSPENDED:
+
+                            continue
+
+                        if pipe.state == states.INITIAL:
+
+                            # Set state of pipeline to SCHEDULING if it is
+                            # in INITIAL.
+                            local_transition(obj=pipe,
+                                             obj_type='Pipeline',
+                                             new_state=states.SCHEDULING,
+                                             profiler=self._prof,
+                                             logger=self._logger,
+                                             reporter=self._report)
+
+                        # Get the next stage of this pipeline to process
+                        exec_stage = pipe.stages[pipe.current_stage - 1]
+
+                        if not exec_stage.uid:
+                            # TODO: Move parent uid, name assignment to
+                            # assign_uid function
```

Comment:
Isn't the code below already doing what the `TODO` requests?

Reply:
Well, I want to move the parent uid assignment to the _assign_uid function as well.
Currently, the _assign_uid function only assigns parent uids to contents (tasks in this 
case) of the object (stage), not the object itself.